app/Jobs/Api/AddTags.php

```diff
@@ -51,54 +53,20 @@ public function handle ()
 
         $photo = Photo::find($this->request['photo_id']);
 
-        $schema = LitterTags::INSTANCE()->getDecodedJSON();
+        $tags = $this->request['litter'];
 
-        $litterTotal = 0;
-        foreach ($this->request['litter'] as $category => $items)
-        {
-            foreach ($items as $column => $quantity)
-            {
-                // Column on photos table to make a relationship with current category eg smoking_id
-                $id_table = $schema->$category->id_table;
-
-                // Full class path
-                $class = 'App\\Models\\Litter\\Categories\\'.$schema->$category->class;
-
-                // Create reference to category.$id_table on photos if it does not exist
-                if (is_null($photo->$id_table))
-                {
-                    $row = $class::create();
-                    $photo->$id_table = $row->id;
-                    $photo->save();
-                }
-
-                // If it does exist, get it
-                else $row = $class::find($photo->$id_table);
-
-                // Update quantity on the category table
-                $row->$column = $quantity;
-                $row->save();
-
-                // Update Leaderboards if user has public privacy settings
-                // todo - save data per leaderboard
-                if (($user->show_name) || ($user->show_username))
-                {
-                    $country = Country::find($photo->country_id);
-                    $state = State::find($photo->state_id);
-                    $city = City::find($photo->city_id);
-                    Redis::zadd($country->country.':Leaderboard', $user->xp, $user->id);
-                    Redis::zadd($country->country.':'.$state->state.':Leaderboard', $user->xp, $user->id);
-                    Redis::zadd($country->country.':'.$state->state.':'.$city->city.':Leaderboard', $user->xp, $user->id);
-                }
-
-                $litterTotal += $quantity;
-            }
-        }
+        /** @var AddTagsToPhotoAction $addTagsAction */
+        $addTagsAction = app(AddTagsToPhotoAction::class);
```

Comment:
Using app() is also an injection, behind the scenes it's doing the same thing as if we inject the object from a constructor or method, and it's pretty common. I did not touch the Jobs constructors, because I think we would be better of removing this one entirely. Why is it a job if it is a process that needs to happen immediately?

Reply:
When we upload from the app, we don't need this to happen immediately. We want to let the user upload as quickly as possible and then we can handle it when the app is ready

Reply:
These operations aren't that slow actually, they don't even span a few ms. Have you noticed any delays?

Reply:
I haven't noticed any delays, but thought it would make more sense to accept the data and let the user continue